src/test_lib_json/main.cpp

```diff
@@ -3903,6 +3903,28 @@ JSONTEST_FIXTURE_LOCAL(FuzzTest, fuzzDoesntCrash) {
                              example.size()));
 }
 
+struct GetLocationLineAndColumnTest : JsonTest::TestCase {
+  void testLineAndColumn(const std::string& doc, ptrdiff_t location, int row,
+                         int column) {
+    int actualRow;
+    int actualColumn;
+    Json::getLocationLineAndColumn(doc.data(), doc.data() + doc.length(),
+                                   location, actualRow, actualColumn);
+    JSONTEST_ASSERT_EQUAL(row, actualRow);
+    JSONTEST_ASSERT_EQUAL(column, actualColumn);
+  }
+};
+
+JSONTEST_FIXTURE_LOCAL(GetLocationLineAndColumnTest, test) {
+  const std::string example = "line 1\nline 2\r\nline 3\rline 4";
```

Comment:
This is not enough testing. There's only one input here.
There should be an exhaustive loop checking the results of every position.
This input only has one kind of newline.